CMakeLists.txt

```diff
@@ -0,0 +1,48 @@
+﻿cmake_minimum_required(VERSION 3.16)
+
+set(CMAKE_GENERATOR_PLATFORM win32)
+
+project(gMod)
+
+if(NOT(CMAKE_SIZEOF_VOID_P EQUAL 4))
+    message(FATAL_ERROR "You are configuring a non 32-bit build, this is not supported. Run cmake with `-A Win32`")
+endif()
+
+set(VERSION "1.0")
+
+set(CMAKE_RUNTIME_OUTPUT_DIRECTORY "${PROJECT_SOURCE_DIR}/bin/")
+set(CMAKE_CXX_STANDARD 23)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+set(CMAKE_MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>")
+
+set_property(GLOBAL PROPERTY USE_FOLDERS OFF)
+
+add_compile_options(/MP /permissive-)
+
+find_library(D3D9_LIB NAMES d3d9 PATHS $ENV{DXSDK_DIR}/lib/x86 NO_DEFAULT_PATH)
```

Comment:
Please here keep the fallback logic. It's used when DXSDK_DIR doesn't resolve the lib paths.
The CI & CD pipelines were having trouble with resolving the paths with the DXSDK_DIR so I resorted to copying the lib and header files in /header and /Lib in order to build